DeskReserve/Controllers/DesksController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using DeskReserve.Data.DBContext;
+using DeskReserve.Data.DBContext.Entity;
+
+namespace DeskReserve.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DesksController : ControllerBase
+    {
+        private readonly ApplicationDbContext _context;
+
+        public DesksController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Desks
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Desk>>> GetDesk()
+        {
+            return await _context.Desks.ToListAsync();
+        }
+
+        // GET: api/Desks/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Desk>> GetDesk(Guid id)
+        {
+            var desk = await _context.Desks.FindAsync(id);
+
+            if (desk == null)
+            {
+                return NotFound();
+            }
+
+            return desk;
+        }
+
+        // PUT: api/Desks/5
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutDesk(Guid id, Desk desk)
+        {
+            if (id != desk.DeskId)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(desk).State = EntityState.Modified;
```

Comment:
Do we really need to explicitly check the state of the object in the DB context to know it is modified?